Makefile

```diff
@@ -410,9 +410,15 @@ include prebuilds.mk
 -include $(filter %.d,$(STDOBJS:%.o=%.d))
 -include $(filter %.d,$(HVLOGOBJS:%.o=%.d))
 
+# 'make all' calculates the current checksum of all .h and .hpp files, storing the checksum in a file. Then it decides whether to run 'make clean' or 'make standard' based on whether any .h and .hpp files have been altered
+HEADER_CHECKSUM_FILE=.header_checksum
+
 all:
-	@$(MAKE) clean
-	@$(MAKE) standard
+	@current_checksum=$$(find . -type f \( -name "*.h" -o -name "*.hpp" \) -print0 | sort -z | xargs -0 sha256sum | sha256sum | awk '{print $$1}'); \
+	if [ ! -f $(HEADER_CHECKSUM_FILE) ] || [ "$$(cat $(HEADER_CHECKSUM_FILE))" != "$$current_checksum" ]; then \
+		$(MAKE) clean; \
+	fi; \
+	$(MAKE) standard && echo "$$current_checksum" > $(HEADER_CHECKSUM_FILE)
```

Comment:
Ignores a -j parameter (or any other parameter) given to the original `make all` I guess.

Reply:
Surprisingly, it doesn't ignore it. I don't know how this stuff works but it's correctly passing the parameter. `make -j8 all` is way faster than `make all`, from 2 minutes down to 30 sec, same as `make standard` and `make -j8 standard` for me.